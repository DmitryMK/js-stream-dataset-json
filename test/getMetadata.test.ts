import DatasetJson from "../src/index";

test("Get metadata", async () => {
  const filePath = "test/data/adsl.json";

  const data = new DatasetJson(filePath);
  await data.init();
  const metadata = await data.getMetadata();
  expect(metadata).toMatchSnapshot();
});
