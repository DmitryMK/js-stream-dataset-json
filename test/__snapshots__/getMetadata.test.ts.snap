// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get metadata 1`] = `
{
  "dataMetadata": {
    "metaDataVersionOID": "CDISC.ADaM.2.1",
    "studyOID": "CDISCPILOT01",
  },
  "itemGroupMetadata": {
    "items": [
      {
        "OID": "ITEMGROUPDATASEQ",
        "label": "Record Identifier",
        "name": "ITEMGROUPDATASEQ",
        "type": "integer",
      },
      {
        "OID": "ADSL.STUDYID",
        "label": "Study Identifier",
        "length": 12,
        "name": "STUDYID",
        "type": "string",
      },
      {
        "OID": "ADSL.USUBJID",
        "label": "Unique Subject Identifier",
        "length": 11,
        "name": "USUBJID",
        "type": "string",
      },
      {
        "OID": "ADSL.SUBJID",
        "label": "Subject Identifier for the Study",
        "length": 4,
        "name": "SUBJID",
        "type": "string",
      },
      {
        "OID": "ADSL.SITEID",
        "label": "Study Site Identifier",
        "length": 3,
        "name": "SITEID",
        "type": "string",
      },
      {
        "OID": "ADSL.SITEGR1",
        "label": "Pooled Site Group 1",
        "length": 3,
        "name": "SITEGR1",
        "type": "string",
      },
      {
        "OID": "ADSL.ARM",
        "label": "Description of Planned Arm",
        "length": 20,
        "name": "ARM",
        "type": "string",
      },
      {
        "OID": "ADSL.TRT01P",
        "label": "Planned Treatment for Period 01",
        "length": 20,
        "name": "TRT01P",
        "type": "string",
      },
      {
        "OID": "ADSL.TRT01PN",
        "label": "Planned Treatment for Period 01 (N)",
        "length": 8,
        "name": "TRT01PN",
        "type": "integer",
      },
      {
        "OID": "ADSL.TRT01A",
        "label": "Actual Treatment for Period 01",
        "length": 20,
        "name": "TRT01A",
        "type": "string",
      },
      {
        "OID": "ADSL.TRT01AN",
        "label": "Actual Treatment for Period 01 (N)",
        "length": 8,
        "name": "TRT01AN",
        "type": "integer",
      },
      {
        "OID": "ADSL.TRTSDT",
        "label": "Date of First Exposure to Treatment",
        "length": 8,
        "name": "TRTSDT",
        "type": "integer",
      },
      {
        "OID": "ADSL.TRTEDT",
        "label": "Date of Last Exposure to Treatment",
        "length": 8,
        "name": "TRTEDT",
        "type": "integer",
      },
      {
        "OID": "ADSL.TRTDUR",
        "label": "Duration of Treatment (days)",
        "length": 8,
        "name": "TRTDUR",
        "type": "integer",
      },
      {
        "OID": "ADSL.AVGDD",
        "label": "Avg Daily Dose (as planned)",
        "length": 8,
        "name": "AVGDD",
        "type": "float",
      },
      {
        "OID": "ADSL.CUMDOSE",
        "label": "Cumulative Dose (as planned)",
        "length": 8,
        "name": "CUMDOSE",
        "type": "float",
      },
      {
        "OID": "ADSL.AGE",
        "label": "Age",
        "length": 8,
        "name": "AGE",
        "type": "integer",
      },
      {
        "OID": "ADSL.AGEGR1",
        "label": "Pooled Age Group 1",
        "length": 5,
        "name": "AGEGR1",
        "type": "string",
      },
      {
        "OID": "ADSL.AGEGR1N",
        "label": "Pooled Age Group 1 (N)",
        "length": 8,
        "name": "AGEGR1N",
        "type": "integer",
      },
      {
        "OID": "ADSL.AGEU",
        "label": "Age Units",
        "length": 5,
        "name": "AGEU",
        "type": "string",
      },
      {
        "OID": "ADSL.RACE",
        "label": "Race",
        "length": 32,
        "name": "RACE",
        "type": "string",
      },
      {
        "OID": "ADSL.RACEN",
        "label": "Race (N)",
        "length": 8,
        "name": "RACEN",
        "type": "integer",
      },
      {
        "OID": "ADSL.SEX",
        "label": "Sex",
        "length": 1,
        "name": "SEX",
        "type": "string",
      },
      {
        "OID": "ADSL.ETHNIC",
        "label": "Ethnicity",
        "length": 22,
        "name": "ETHNIC",
        "type": "string",
      },
      {
        "OID": "ADSL.SAFFL",
        "label": "Safety Population Flag",
        "length": 1,
        "name": "SAFFL",
        "type": "string",
      },
      {
        "OID": "ADSL.ITTFL",
        "label": "Intent-To-Treat Population Flag",
        "length": 1,
        "name": "ITTFL",
        "type": "string",
      },
      {
        "OID": "ADSL.EFFFL",
        "label": "Efficacy Population Flag",
        "length": 1,
        "name": "EFFFL",
        "type": "string",
      },
      {
        "OID": "ADSL.COMP8FL",
        "label": "Completers of Week 8 Population Flag",
        "length": 1,
        "name": "COMP8FL",
        "type": "string",
      },
      {
        "OID": "ADSL.COMP16FL",
        "label": "Completers of Week 16 Population Flag",
        "length": 1,
        "name": "COMP16FL",
        "type": "string",
      },
      {
        "OID": "ADSL.COMP24FL",
        "label": "Completers of Week 24 Population Flag",
        "length": 1,
        "name": "COMP24FL",
        "type": "string",
      },
      {
        "OID": "ADSL.DISCONFL",
        "label": "Did the Subject Discontinue the Study?",
        "length": 1,
        "name": "DISCONFL",
        "type": "string",
      },
      {
        "OID": "ADSL.DSRAEFL",
        "label": "Discontinued due to AE?",
        "length": 1,
        "name": "DSRAEFL",
        "type": "string",
      },
      {
        "OID": "ADSL.DTHFL",
        "label": "Subject Died?",
        "length": 1,
        "name": "DTHFL",
        "type": "string",
      },
      {
        "OID": "ADSL.BMIBL",
        "label": "Baseline BMI (kg/m^2)",
        "length": 8,
        "name": "BMIBL",
        "type": "float",
      },
      {
        "OID": "ADSL.BMIBLGR1",
        "label": "Pooled Baseline BMI Group 1",
        "length": 6,
        "name": "BMIBLGR1",
        "type": "string",
      },
      {
        "OID": "ADSL.HEIGHTBL",
        "label": "Baseline Height (cm)",
        "length": 8,
        "name": "HEIGHTBL",
        "type": "float",
      },
      {
        "OID": "ADSL.WEIGHTBL",
        "label": "Baseline Weight (kg)",
        "length": 8,
        "name": "WEIGHTBL",
        "type": "float",
      },
      {
        "OID": "ADSL.EDUCLVL",
        "label": "Years of Education",
        "length": 8,
        "name": "EDUCLVL",
        "type": "integer",
      },
      {
        "OID": "ADSL.DISONSDT",
        "label": "Date of Onset of Disease",
        "length": 8,
        "name": "DISONSDT",
        "type": "integer",
      },
      {
        "OID": "ADSL.DURDIS",
        "label": "Duration of Disease (Months)",
        "length": 8,
        "name": "DURDIS",
        "type": "float",
      },
      {
        "OID": "ADSL.DURDSGR1",
        "label": "Pooled Disease Duration Group 1",
        "length": 4,
        "name": "DURDSGR1",
        "type": "string",
      },
      {
        "OID": "ADSL.VISIT1DT",
        "label": "Date of Visit 1",
        "length": 8,
        "name": "VISIT1DT",
        "type": "integer",
      },
      {
        "OID": "ADSL.RFSTDTC",
        "label": "Subject Reference Start Date/Time",
        "length": 20,
        "name": "RFSTDTC",
        "type": "string",
      },
      {
        "OID": "ADSL.RFENDTC",
        "label": "Subject Reference End Date/Time",
        "length": 20,
        "name": "RFENDTC",
        "type": "string",
      },
      {
        "OID": "ADSL.VISNUMEN",
        "label": "End of Trt Visit (Vis 12 or Early Term.)",
        "length": 8,
        "name": "VISNUMEN",
        "type": "integer",
      },
      {
        "OID": "ADSL.RFENDT",
        "label": "Date of Discontinuation/Completion",
        "length": 8,
        "name": "RFENDT",
        "type": "integer",
      },
      {
        "OID": "ADSL.DCDECOD",
        "label": "Standardized Disposition Term",
        "length": 27,
        "name": "DCDECOD",
        "type": "string",
      },
      {
        "OID": "ADSL.DCREASCD",
        "label": "Reason for Discontinuation",
        "length": 18,
        "name": "DCREASCD",
        "type": "string",
      },
      {
        "OID": "ADSL.MMSETOT",
        "label": "MMSE Total",
        "length": 8,
        "name": "MMSETOT",
        "type": "integer",
      },
    ],
    "label": "Subject-Level Analysis",
    "name": "ADSL",
    "records": 254,
  },
}
`;
